src/ga/population.rs

```diff
@@ -9,25 +10,49 @@ use super::{individual::Chromosome, Individual};
 /// Implement this trait in order to provide custom population generator
 /// and feed it to an solver.
 pub trait PopulationGenerator<T: Chromosome> {
-  fn generate(&self, count: usize) -> Vec<Individual<T>>;
+  fn generate(&mut self, count: usize) -> Vec<Individual<T>>;
 }
 
 /// Implements [PopulationGenerator] trait. Can be used with genetic algorithm.
 ///
 /// Generates vector of random points from R^(dim) space within passed domain constraints.
-pub struct RandomPoints {
+pub struct RandomPoints<R: Rng> {
   dim: usize,
   constraints: Vec<(f64, f64)>,
+  rng: R,
 }
 
-impl RandomPoints {
-  /// Returns [RandomPoints] population generator with given constraints
+impl RandomPoints<ThreadRng> {
+  /// Returns [RandomPoints] population generator with given constraints and default RNG
   ///
   /// ### Arguments
   ///
   /// * `dim` -- Dimension of the sampling space
   /// * `constraints` -- Ranges for coordinates
   pub fn with_constraints(dim: usize, constraints: Vec<Range<f64>>) -> Self {
+    Self::with_constraints_and_rng(dim, constraints, thread_rng())
+  }
+
+  /// Returns [RandomPoints] population generator with no explicit constraints and default RNG.
+  /// Points coords will be from range 0.0..1.0.
+  ///
+  /// ### Arguments
+  ///
+  /// * `dim` -- Dimension of the sampling space
+  pub fn new(dim: usize) -> Self {
+    Self::wih_rng(dim, thread_rng())
```

Comment:
Typo 
```suggestion
    Self::wih_rng(dim, thread_rng())
```
```suggestion
    Self::with_rng(dim, thread_rng())
```

possibly the method should be renamed